cookbook/form/data_transformers.rst

```diff
@@ -4,29 +4,143 @@
 How to Use Data Transformers
 ============================
 
-You'll often find the need to transform the data the user entered in a form into
-something else for use in your program. You could easily do this manually in your
-controller, but what if you want to use this specific form in different places?
-
-Say you have a one-to-one relation of Task to Issue, e.g. a Task optionally has an
-issue linked to it. Adding a listbox with all possible issues can eventually lead to
-a really long listbox in which it is impossible to find something. You might
-want to add a textbox instead, where the user can simply enter the issue number.
-
-You could try to do this in your controller, but it's not the best solution.
-It would be better if this issue were automatically converted to an Issue object.
-This is where Data Transformers come into play.
+Data transformers are used to translate the data for a field into a format that can
+be displayed in a form (and back on submit). They're already used internally for
+many field types. For example, the :doc:`date field type </reference/forms/types/date>`
+can be rendered as a ``yyyy-MM-dd``-formatted input textbox. Internally, a data transformer
+converts the starting ``DateTime`` value of the field into the ``yyyy-MM-dd`` string
+to render the form, and then back into a ``DateTime`` object on submit.
 
 .. caution::
 
     When a form field has the ``inherit_data`` option set, Data Transformers
     won't be applied to that field.
 
+Simple Example: Sanitizing HTML on User Input
+---------------------------------------------
+
+Suppose you have a Task form with a description ``textarea`` type::
+
+    // src/AppBundle/Form/TaskType.php
+    // ...
+
+    class TaskType extends AbstractType
+    {
+        public function buildForm(FormBuilderInterface $builder, array $options)
+        {
+            $builder
+                ->add('description', 'textarea');
+        }
+        
+        public function setDefaultOptions(OptionsResolverInterface $resolver)
+        {
+            $resolver->setDefaults(array(
+                'data_class' => 'AppBundle\Entity\Task'
+            ));
+        }
+
+        // ...
+    }
+
+But, there are two complications:
+
+#. Your users are allowed to use *some* HTML tags, but not others: you need a way
+   to call :phpfunction:`striptags` after the form is submitted.
+
+#. To be friendly, you want to convert ``<br/>`` tags into line breaks (``\n``) before
+   rendering the field so the text is easier to edit.
```

Comment:
I could argue (at least for the line break thing) that we're not changing the data - `\n` versus `<br/>` is sort of a "formatting" thing.

Anyways, I'm not willing to remove this first nice simple example. But, I would definitely be open to a different example - I was honestly thinking pretty hard to get something "real", but understandable. Can you think of a better example, maybe from some real problem you've seen someone tackling?